src/Symfony/Component/Translation/PluralizationRules.php

```diff
@@ -25,13 +25,15 @@ class PluralizationRules
     /**
      * Returns the plural position to use for the given locale and number.
      *
-     * @param int    $number The number
+     * @param float  $number The number
      * @param string $locale The locale
      *
      * @return int The plural position
      */
     public static function get($number, $locale/*, bool $triggerDeprecation = true*/)
     {
+        $number = abs($number);
```

Comment:
Is that something that works for all languages? I.e. could it be that there are languages where a number is treated differently when either being positive or negative?

Reply:
A sure thing is majority of the languages handle it the same way and currently they get wrong results. as the rules assume having a positive number. This will actually ensure that the assertion taken until then is met.

You can see ignoring the sign is the path taken by CLDR:
http://cldr.unicode.org/index/cldr-spec/plural-rules

If there is different rules for negative numbers, we can still add them at this moment: create `$absoluteNumber = abs($number)` so a given language can use the absolute number or the signed one.

But:
- We sure have negative plural broken for most languages and get it fixed with this change.
- If there is edge-cases, they very unlikely match the current implementation (they have way more chances to be broken too because rules don't take it into account)

Reply:
To get an overview, we could grep all languages from angular/angular for instance that have a `Math.abs()` in their plural rule:
https://github.com/angular/angular/blob/master/packages/common/locales/fr.ts#L15
https://github.com/angular/angular/blob/master/packages/common/locales/en.ts#L15

Actually by looking at some random I couldn't find any that don't have it but I will extract an exhaustive list from it.

Reply:
Angular rules parses CLDR data to get their rules and this input data is:
https://github.com/papandreou/node-cldr/blob/master/3rdparty/cldr/common/supplemental/plurals.xml

Where `i` is `floor(abs(n))` so first all rules based on `i` are guaranteed to be symmetric and this is the majority of them. I'm investigating on other ones because I identified yet very probable mistakes:

- wae "Walser" (Suisse) => -1 should be singular
- vo "Volapük" => It comes from German so -1 is likely singular too

So I think most of the rules here using `n` (the raw number) use it to handle properly the decimal part while it does not take the sign into account and would just output the "other" form for every negative number as a fallback for unknown rule.